core/src/main/java/ch/cyberduck/core/shared/DefaultPathHomeFeature.java

```diff
@@ -18,31 +18,44 @@
  * feedback@cyberduck.ch
  */
 
-import ch.cyberduck.core.Host;
 import ch.cyberduck.core.Path;
 import ch.cyberduck.core.PathNormalizer;
+import ch.cyberduck.core.Session;
+import ch.cyberduck.core.TildePathExpander;
 import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.features.Home;
 
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.EnumSet;
+
 public class DefaultPathHomeFeature extends AbstractHomeFeature {
     private static final Logger log = LogManager.getLogger(DefaultPathHomeFeature.class);
 
-    private final Host host;
+    private final Session<?> session;
 
-    public DefaultPathHomeFeature(final Host host) {
-        this.host = host;
+    public DefaultPathHomeFeature(final Session<?> session) {
+        this.session = session;
     }
 
     @Override
     public Path find() throws BackgroundException {
-        if(StringUtils.isNotBlank(host.getDefaultPath())) {
-            return PathNormalizer.compose(ROOT, host.getDefaultPath());
+        if(StringUtils.isNotBlank(session.getHost().getDefaultPath())) {
+            if(StringUtils.startsWith(session.getHost().getDefaultPath(), Path.HOME)) {
```

Comment:
`~myfolder/subfolder` should not expand but `~/myfolder` and `~`.

Reply:
Relates to #14590.